worlds/AutoWorld.py

```diff
@@ -3,16 +3,18 @@
 import hashlib
 import logging
 import pathlib
-import random
```

Comment:
line 331 uses this

I'm not sure why it's in the `if TYPE_CHECKING` block.

Reply:
Out of scope for this PR, but it would be better if it's `from random import Random` to discourage any use of the global `random` state.

Reply:
I brought in `from random import Random` and used it in the few places it was directly referenced in AutoWorld.